<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="netboxr">
<title>NetBoxR Tutorial • netboxr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="NetBoxR Tutorial">
<meta property="og:description" content="netboxr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">netboxr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/netboxrTutorial.html">NetBoxR Tutorial</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="netboxrTutorial_files/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="netboxrTutorial_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="netboxrTutorial_files/datatables-binding-0.27/datatables.js"></script><link href="netboxrTutorial_files/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="netboxrTutorial_files/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="netboxrTutorial_files/dt-core-1.12.1/js/jquery.dataTables.min.js"></script><link href="netboxrTutorial_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="netboxrTutorial_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>NetBoxR Tutorial</h1>
                        <h4 data-toc-skip class="author">Eric Minwei Liu
and Augustin Luna</h4>
            
            <h4 data-toc-skip class="date">19 May, 2023</h4>
      
      
      <div class="d-none name"><code>netboxrTutorial.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <strong>netboxr</strong> package composes a number of functions
to retrive and process genetic data from large-scale genomics projects
(e.g. TCGA projects) including from mutations, copy number alterations,
gene expression and DNA methylation. The netboxr package implements
NetBox algorithm in R package. NetBox algorithm integrates genetic
alterations with literature-curated pathway knowledge to identify
pathway modules in cancer. NetBox algorithm uses (1) global network null
model and (2) local network null model to access the statistic
significance of the discovered pathway modules.</p>
</div>
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"netboxr"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h3>
<p>Load <strong>netboxr</strong> package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mil2041.github.io/netboxr/">netboxr</a></span><span class="op">)</span></span></code></pre></div>
<p>A list of all accessible vignettes and methods is available with the
following command:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"netboxr"</span><span class="op">)</span></span></code></pre></div>
<p>For help on any <strong>netboxr</strong> package functions, use one
of the following command formats:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">geneConnector</span><span class="op">)</span></span>
<span><span class="op">?</span><span class="va">geneConnector</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="example-of-cerami-et-al--plos-one-2010">Example of Cerami et al. PLoS One 2010<a class="anchor" aria-label="anchor" href="#example-of-cerami-et-al--plos-one-2010"></a>
</h2>
<p>This is an example to reproduce the network discovered on <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0008918" class="external-link">Cerami
et al.(2010)</a>.</p>
<p>The results presented here are comparable to the those from Cerami et
al. 2010 though the unadjusted p-values for linker genes are not the
same. It is because the unadjusted p-value of linker genes in Cerami et
al. 2010 were calculated by the probabiliy of the observed data point,
Pr(X). The netboxr used the probability of an observed or more extreme
assuming the null hypothesis is true, Pr(X&gt;=x|H), as unadjusted
p-value for linker genes. The final number of linker genes after FDR
correction are the same between netboxr result and original Cerami et
al. 2010.</p>
<div class="section level3">
<h3 id="load-human-interactions-network-hin-network">Load Human Interactions Network (HIN) network<a class="anchor" aria-label="anchor" href="#load-human-interactions-network-hin-network"></a>
</h3>
<p>Load pre-defined HIN network and simplify the interactions by
removing loops and duplicated interactions in the network. The netowork
after reduction contains 9264 nodes and 68111 interactions.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">netbox2010</span><span class="op">)</span></span>
<span><span class="va">sifNetwork</span> <span class="op">&lt;-</span> <span class="va">netbox2010</span><span class="op">$</span><span class="va">network</span></span>
<span><span class="va">graphReduced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/networkSimplify.html">networkSimplify</a></span><span class="op">(</span><span class="va">sifNetwork</span>, directed <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading network of 9264 nodes and 157780 interactions</span></span></code></pre>
<pre><code><span><span class="co">## Treated as undirected network</span></span></code></pre>
<pre><code><span><span class="co">## Removing multiple interactions and loops</span></span></code></pre>
<pre><code><span><span class="co">## Returning network of 9264 nodes and 68111 interactions</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="load-altered-gene-list">Load altered gene list<a class="anchor" aria-label="anchor" href="#load-altered-gene-list"></a>
</h3>
<p>The altered gene list contains 517 candidates from mutations and copy
number alterations.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">netbox2010</span><span class="op">$</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 517</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="map-altered-gene-list-on-hin-network">Map altered gene list on HIN network<a class="anchor" aria-label="anchor" href="#map-altered-gene-list-on-hin-network"></a>
</h3>
<p>The geneConnector function in the netboxr package takes altered gene
list as input and maps the genes on the curated network to find the
local processes represented by the gene list.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Use Benjamini-Hochberg method to do multiple hypothesis correction for</span></span>
<span><span class="co">## linker candidates.</span></span>
<span></span>
<span><span class="co">## Use edge-betweeness method to detect community structure in the network.</span></span>
<span><span class="va">threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/geneConnector.html">geneConnector</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">geneList</span>, networkGraph <span class="op">=</span> <span class="va">graphReduced</span>, directed <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    pValueAdj <span class="op">=</span> <span class="st">"BH"</span>, pValueCutoff <span class="op">=</span> <span class="va">threshold</span>, communityMethod <span class="op">=</span> <span class="st">"ebc"</span>, keepIsolatedNodes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 274 / 517 candidate nodes match the name in the network of 9264 </span></span>
<span><span class="co">##                 nodes</span></span></code></pre>
<pre><code><span><span class="co">## Only test neighbor nodes with local degree equals or exceeds 2</span></span></code></pre>
<pre><code><span><span class="co">## Multiple hypothesis corrections for 892 neighbor nodes in the network</span></span></code></pre>
<pre><code><span><span class="co">## For p-value 0.05 cut-off, 6 nodes were included as linker nodes</span></span></code></pre>
<pre><code><span><span class="co">## Connecting 274 candidate nodes and 6 linker nodes</span></span></code></pre>
<pre><code><span><span class="co">## Remove 208 isolated candidate nodes from the input</span></span></code></pre>
<pre><code><span><span class="co">## Final network contains 72 nodes and 152 interactions</span></span></code></pre>
<pre><code><span><span class="co">## Detecting modules using "edge betweeness" method</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add edge annotations</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="va">edges</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">netboxOutput</span></span>
<span><span class="va">interactionType</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">edges</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">interactionTypeColor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">interactionType</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">"Spectral"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">edgeColors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">interactionType</span>, <span class="va">interactionTypeColor</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">edgeColors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"INTERACTION_TYPE"</span>, <span class="st">"COLOR"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">netboxGraphAnnotated</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotateGraph.html">annotateGraph</a></span><span class="op">(</span>netboxResults <span class="op">=</span> <span class="va">results</span>, edgeColors <span class="op">=</span> <span class="va">edgeColors</span>,</span>
<span>    directed <span class="op">=</span> <span class="cn">FALSE</span>, linker <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the p-value of the selected linker</span></span>
<span><span class="va">linkerDF</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">neighborData</span></span>
<span><span class="va">linkerDF</span><span class="op">[</span><span class="va">linkerDF</span><span class="op">$</span><span class="va">pValueFDR</span> <span class="op">&lt;</span> <span class="va">threshold</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##         idx   name localDegree globalDegree    pValueRaw oddsRatio  pValueFDR</span></span>
<span><span class="co">## CRK    1712    CRK          11           81 2.392088e-05  1.708732 0.01866731</span></span>
<span><span class="co">## IFNAR1 4546 IFNAR1           6           23 4.185496e-05  2.518726 0.01866731</span></span>
<span><span class="co">## CBL      20    CBL          14          140 6.505470e-05  1.361057 0.01934293</span></span>
<span><span class="co">## GAB1    500   GAB1           8           57 2.483197e-04  1.751122 0.04887827</span></span>
<span><span class="co">## CDK6    414   CDK6           5           21 3.008515e-04  2.406906 0.04887827</span></span>
<span><span class="co">## PTPN11   84 PTPN11          14          163 3.287776e-04  1.191405 0.04887827</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The geneConnector function returns a list of data frames.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "netboxGraph"      "netboxCommunity"  "netboxOutput"     "nodeType"        </span></span>
<span><span class="co">## [5] "moduleMembership" "neighborData"</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot graph with the Fruchterman-Reingold layout algorithm As an example, plot</span></span>
<span><span class="co"># both the original and the annotated graphs Save the layout for easier</span></span>
<span><span class="co"># comparison</span></span>
<span><span class="va">graph_layout</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/layout_with_fr.html" class="external-link">layout_with_fr</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">netboxGraph</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the original graph</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">netboxCommunity</span>, <span class="va">results</span><span class="op">$</span><span class="va">netboxGraph</span>, layout <span class="op">=</span> <span class="va">graph_layout</span><span class="op">)</span></span></code></pre></div>
<p><img src="netboxrTutorial_files/figure-html/netboxrExampleGeneConnector-1.png" width="1152" style="display:block; margin: auto"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the edge annotated graph</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">netboxCommunity</span>, <span class="va">netboxGraphAnnotated</span>, layout <span class="op">=</span> <span class="va">graph_layout</span>, vertex.size <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    vertex.shape <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">netboxGraphAnnotated</span><span class="op">)</span><span class="op">$</span><span class="va">shape</span>, edge.color <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/E.html" class="external-link">E</a></span><span class="op">(</span><span class="va">netboxGraphAnnotated</span><span class="op">)</span><span class="op">$</span><span class="va">interactionColor</span>,</span>
<span>    edge.width <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add interaction type annotations</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"bottomleft"</span>, legend <span class="op">=</span> <span class="va">interactionType</span>, col <span class="op">=</span> <span class="va">interactionTypeColor</span>, lty <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    lwd <span class="op">=</span> <span class="fl">2</span>, bty <span class="op">=</span> <span class="st">"n"</span>, cex <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="netboxrTutorial_files/figure-html/netboxrExampleGeneConnector-2.png" width="1152" style="display:block; margin: auto"></p>
</div>
<div class="section level3">
<h3 id="consistency-with-previously-published-results">Consistency with Previously Published Results<a class="anchor" aria-label="anchor" href="#consistency-with-previously-published-results"></a>
</h3>
<p>The GBM result by netboxr identified exactly the same linker genes (6
linker genes), the same number of modules (10 modules) and the same
genes in each identified module as GBM result in Cerami et al. 2010.</p>
<p>The results of netboxr are consistent with previous implementation of
the NetBox algorithm. The RB1 and PIK3R1 modules are clearly represented
in the figure. For example, the RB1 module contains genes in blue color
and enclosed by light orange circle. The PIK3R1 module contains genes in
orange color and enclosed by pink circle.</p>
</div>
</div>
<div class="section level2">
<h2 id="statistical-significance-of-discovered-network">Statistical Significance of Discovered Network<a class="anchor" aria-label="anchor" href="#statistical-significance-of-discovered-network"></a>
</h2>
<p>NetBox algorithm used (1) global network null model and (2) local
network null model to access the statistical significance of the
discovered network.</p>
<div class="section level3">
<h3 id="global-network-null-model">Global Network Null Model<a class="anchor" aria-label="anchor" href="#global-network-null-model"></a>
</h3>
<p>The global network null model calculates the empirical p-value as the
number of times (over a set of iterations) the size of the largest
connected component (the giant component) in the network coming from the
same number of randomly selected genes (number of genes is 274 in this
example) equals or exceeds the size of the largest connected component
in the observed network. The suggested iterations are 1000.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This function will need a lot of time to complete.</span></span>
<span><span class="va">globalTest</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/globalNullModel.html">globalNullModel</a></span><span class="op">(</span>netboxGraph <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">netboxGraph</span>, networkGraph <span class="op">=</span> <span class="va">graphReduced</span>,</span>
<span>    iterations <span class="op">=</span> <span class="fl">10</span>, numOfGenes <span class="op">=</span> <span class="fl">274</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="local-network-null-model">Local Network Null Model<a class="anchor" aria-label="anchor" href="#local-network-null-model"></a>
</h3>
<p>Local network null model evaluates the deviation of modularity in the
observed network from modularity distribution in the random network. For
each interaction, a random network is produced from local re-wiring of
literature curated network. It means all nodes in the network kept the
same degree of connections but connect to new neighbors randomly.
Suggested iterations is 1000.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">localTest</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/localNullModel.html">localNullModel</a></span><span class="op">(</span>netboxGraph <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">netboxGraph</span>, iterations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ###########</span></span></code></pre>
<pre><code><span><span class="co">## Based on 1000 random trails</span></span></code></pre>
<pre><code><span><span class="co">## Random networks: mean modularity = 0.297</span></span></code></pre>
<pre><code><span><span class="co">## Random networks: sd modularity = 0.059</span></span></code></pre>
<pre><code><span><span class="co">## Observed network modularity is: 0.519</span></span></code></pre>
<pre><code><span><span class="co">## Observed network modularity z-score is: 3.74</span></span></code></pre>
<pre><code><span><span class="co">## One-tail p-value is: 9.185e-05</span></span></code></pre>
<p>Through 1000 iterations, we can obtain the mean and the standard
deviation of modularity in the local network null model. Using the mean
(~0.3) and the standard deviation (0.06), we can covert the observed
modularity in the network (0.519) into a Z-score (~3.8). From the
Z-score, we can calculate one-tail p-value. If one-tail pvalue is less
than 0.05, the observed modularity is significantly different from
random. In the histogram, the blue region is the distribution of
modularity in the local network null model. The red vertical line is the
observed modularity in the NetBox results.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">localTest</span><span class="op">$</span><span class="va">randomModularityScore</span>, breaks <span class="op">=</span> <span class="fl">35</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">h</span><span class="op">$</span><span class="va">density</span> <span class="op">=</span> <span class="va">h</span><span class="op">$</span><span class="va">counts</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">h</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">h</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.1</span><span class="op">)</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.6</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"lightblue"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="va">localTest</span><span class="op">$</span><span class="va">modularityScoreObs</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="netboxrTutorial_files/figure-html/netboxrExampleLocalTestPlot-1.png" width="1152" style="display:block; margin: auto"></p>
<ul>
<li><p>The global null model is used to assess the global connectivity
(number of nodes and edges) of the largest module in the identified
network compared with the same number but randomly selected gene
list.</p></li>
<li><p>The local null model is used to assess the network modularity in
the identified network compared with random re-wired network.</p></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="view-module-membership">View Module Membership<a class="anchor" aria-label="anchor" href="#view-module-membership"></a>
</h2>
<p>The table below shows the module memberships for all genes.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">moduleMembership</span>, rownames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4d5fdcd4b7a909ba4395" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4d5fdcd4b7a909ba4395">{"x":{"filter":"none","vertical":false,"data":[["CBL","ADAM12","PIK3R1","PTPN11","KIT","MPDZ","KDR","ERBB2","SPRY2","FRS2","TEK","GAB1","PDGFRA","IFNG","VLDLR","DOCK1","EGFR","JAK2","CRK","PIK3C2B","PTPRB","PTEN","SH3GL2","FGF23","MAPK11","PIK3CA","HLA-DRA","AVIL","AGAP2","TBP","RB1","TAF1","CDK6","CDKN2A","TP53","HSPA1A","RBBP5","SNAPC3","MDM4","MDM2","KLF6","CDKN2C","CDK4","PIM1","BNC2","EPHA3","CCND2","CDKN2B","IFNW1","IFNA2","IFNB1","IFNAR1","IFNA1","KCNA5","PTPRD","PTPRE","CNTN2","NCAM1","NUP50","THOC4","NUP107","SNRPE","DCTN2","TUBGCP2","FGFR1OP","TUBGCP6","A2M","LYZ","PPARA","STAC3","CCT2","CCT6A"],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,3,3,3,3,3,4,4,4,5,5,6,6,6,6,7,7,7,7,8,8,9,9,10,10]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>geneSymbol<\/th>\n      <th>membership<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level2">
<h2 id="write-netbox-output-to-files">Write NetBox Output to Files<a class="anchor" aria-label="anchor" href="#write-netbox-output-to-files"></a>
</h2>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Write results for further visilaztion in the cytoscape software.  network.sif</span></span>
<span><span class="co"># file is the NetBox algorithm output in SIF format.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">netboxOutput</span>, file <span class="op">=</span> <span class="st">"network.sif"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    col.names <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># netighborList.txt file contains the information of all neighbor nodes.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">neighborData</span>, file <span class="op">=</span> <span class="st">"neighborList.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    col.names <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># community.membership.txt file indicates the identified pathway module</span></span>
<span><span class="co"># numbers.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">moduleMembership</span>, file <span class="op">=</span> <span class="st">"community.membership.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># nodeType.txt file indicates the node is 'linker' node or 'candidate' node.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">nodeType</span>, file <span class="op">=</span> <span class="st">"nodeType.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="term-enrichment-in-modules-using-gene-ontology-go-analysis">Term Enrichment in Modules using Gene Ontology (GO) Analysis<a class="anchor" aria-label="anchor" href="#term-enrichment-in-modules-using-gene-ontology-go-analysis"></a>
</h2>
<p>After module identification, one main task is understanding the
biological processes that may be represented by the returned modules.
Here we use the Bioncoductor clusterProfiler to do an enrichment
analysis using GO Biological Process terms on a selected module.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## clusterProfiler v4.8.1  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use clusterProfiler in published research, please cite:</span></span>
<span><span class="co">## T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'clusterProfiler'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:igraph':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     simplify</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:igraph':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     normalize, path, union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:clusterProfiler':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rename</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'IRanges'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:clusterProfiler':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     slice</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'AnnotationDbi'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:clusterProfiler':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     select</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">module</span> <span class="op">&lt;-</span> <span class="fl">6</span></span>
<span><span class="va">selectedModule</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">moduleMembership</span><span class="op">[</span><span class="va">results</span><span class="op">$</span><span class="va">moduleMembership</span><span class="op">$</span><span class="va">membership</span> <span class="op">==</span></span>
<span>    <span class="va">module</span>, <span class="op">]</span></span>
<span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="va">selectedModule</span><span class="op">$</span><span class="va">geneSymbol</span></span>
<span></span>
<span><span class="co"># Check available ID types in for the org.Hs.eg.db annotation package</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">keytypes</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"</span></span>
<span><span class="co">##  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    </span></span>
<span><span class="co">## [11] "GENETYPE"     "GO"           "GOALL"        "IPI"          "MAP"         </span></span>
<span><span class="co">## [16] "OMIM"         "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"        </span></span>
<span><span class="co">## [21] "PMID"         "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"      </span></span>
<span><span class="co">## [26] "UNIPROT"</span></span></code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/bitr.html" class="external-link">bitr</a></span><span class="op">(</span><span class="va">geneList</span>, fromType <span class="op">=</span> <span class="st">"SYMBOL"</span>, toType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENTREZID"</span><span class="op">)</span>, OrgDb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Warning in bitr(geneList, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb =</span></span>
<span><span class="co">## "org.Hs.eg.db"): 25% of input gene IDs are fail to map...</span></span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ids</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   SYMBOL ENTREZID</span></span>
<span><span class="co">## 1  NUP50    10762</span></span>
<span><span class="co">## 3 NUP107    57122</span></span>
<span><span class="co">## 4  SNRPE     6635</span></span></code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html" class="external-link">enrichGO</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">ids</span><span class="op">$</span><span class="va">ENTREZID</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>    pvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>, qvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, readable <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="enrichment-results">Enrichment Results<a class="anchor" aria-label="anchor" href="#enrichment-results"></a>
</h3>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ego</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                       Description GeneRatio   BgRatio</span></span>
<span><span class="co">## GO:0051028 GO:0051028                    mRNA transport       2/3 130/18614</span></span>
<span><span class="co">## GO:0050657 GO:0050657            nucleic acid transport       2/3 160/18614</span></span>
<span><span class="co">## GO:0050658 GO:0050658                     RNA transport       2/3 160/18614</span></span>
<span><span class="co">## GO:0006606 GO:0006606       protein import into nucleus       2/3 163/18614</span></span>
<span><span class="co">## GO:0051236 GO:0051236 establishment of RNA localization       2/3 163/18614</span></span>
<span><span class="co">## GO:0051170 GO:0051170               import into nucleus       2/3 168/18614</span></span>
<span><span class="co">##                  pvalue    p.adjust       qvalue       geneID Count</span></span>
<span><span class="co">## GO:0051028 0.0001445448 0.001811182 0.0003813015 NUP50/NUP107     2</span></span>
<span><span class="co">## GO:0050657 0.0002190370 0.001811182 0.0003813015 NUP50/NUP107     2</span></span>
<span><span class="co">## GO:0050658 0.0002190370 0.001811182 0.0003813015 NUP50/NUP107     2</span></span>
<span><span class="co">## GO:0006606 0.0002273296 0.001811182 0.0003813015 NUP50/NUP107     2</span></span>
<span><span class="co">## GO:0051236 0.0002273296 0.001811182 0.0003813015 NUP50/NUP107     2</span></span>
<span><span class="co">## GO:0051170 0.0002414910 0.001811182 0.0003813015 NUP50/NUP107     2</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="visualize-enrichment-results">Visualize Enrichment Results<a class="anchor" aria-label="anchor" href="#visualize-enrichment-results"></a>
</h3>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html" class="external-link">dotplot</a></span><span class="op">(</span><span class="va">ego</span><span class="op">)</span></span></code></pre></div>
<p><img src="netboxrTutorial_files/figure-html/unnamed-chunk-3-1.png" width="960" style="display:block; margin: auto"></p>
</div>
</div>
<div class="section level2">
<h2 id="alternative-module-discovery-methods">Alternative Module Discovery Methods<a class="anchor" aria-label="anchor" href="#alternative-module-discovery-methods"></a>
</h2>
<p>In netboxr, we used the Girvan-Newman algorithm
(communityMethod=“ebc”) as the default method to detect community
membership in the identified network. The Girvan-Newman algorithm
iteratativly removes the edge in the network with highest edge
betweeness until no edges left. When the identified network contains
many edges, the Girvan-Newman algorithm will spend a large amount of
time to remove edges and re-calucalte the edge betweenese score in the
network. If the user cannot get the community detection result in
reasonable time, we suggest to switch to leading eigenvector method
(communityMethod=“lec”) for community detection. Users can check
original papers of <a href="http://www.pnas.org/content/99/12/7821" class="external-link">the
Girvan-Newman algorithm</a> and <a href="https://journals.aps.org/pre/abstract/10.1103/PhysRevE.74.036104" class="external-link">leading
eigenvector method</a> for more details.</p>
</div>
<div class="section level2">
<h2 id="alternative-pathway-data">Alternative Pathway Data<a class="anchor" aria-label="anchor" href="#alternative-pathway-data"></a>
</h2>
<div class="section level3">
<h3 id="using-tabular-simple-interaction-format-sif-based-network-data">Using Tabular Simple Interaction Format (SIF)-Based Network
Data<a class="anchor" aria-label="anchor" href="#using-tabular-simple-interaction-format-sif-based-network-data"></a>
</h3>
<p>Users can load alternative pathway data formatted in the SIF format
(Simple Interaction Format). SIF is a space/tab separated format that
summarizes interactions in a graph as an edgelist. In the format, every
row corresponds to an individual interaction (edge) between a source and
a target node. NOTE: An arbitrary interaction type can be used, such as
“interacts” if the true interaction type is unknown.</p>
<pre><code>PARTICIPANT_A INTERACTION_TYPE PARTICIPANT_B
nodeA relationship nodeB
nodeC relationship nodeA
nodeD relationship nodeE</code></pre>
<p>Resources, such as the Functional Interaction network from Reactome
(<a href="https://reactome.org/download-data" class="external-link uri">https://reactome.org/download-data</a>) and StringDB (<a href="https://string-db.org/" class="external-link uri">https://string-db.org/</a>)
provide network information in formats reusable as a SIF. NOTE: The next
section demonstrates how to retrieve SIF-based networks for many
well-known interaction databases using paxtoolsr.</p>
<p>SIF formatted data can be passed to networkSimplify(). The result of
which is used with the geneConnector() function as other examples in
this vignette demonstrate.</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example</span> <span class="op">&lt;-</span> <span class="st">"PARTICIPANT_A\tINTERACTION_TYPE\tPARTICIPANT_B</span></span>
<span><span class="st">TP53\tinteracts\tMDM2</span></span>
<span><span class="st">MDM2\tinteracts\tMDM4"</span></span>
<span></span>
<span><span class="va">sif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span>text <span class="op">=</span> <span class="va">example</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">graphReduced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/networkSimplify.html">networkSimplify</a></span><span class="op">(</span><span class="va">sif</span>, directed <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading network of 3 nodes and 2 interactions</span></span></code></pre>
<pre><code><span><span class="co">## Treated as undirected network</span></span></code></pre>
<pre><code><span><span class="co">## Removing multiple interactions and loops</span></span></code></pre>
<pre><code><span><span class="co">## Returning network of 3 nodes and 2 interactions</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="using-paxtoolsr-for-pathway-commons-data">Using PaxtoolsR for Pathway Commons Data<a class="anchor" aria-label="anchor" href="#using-paxtoolsr-for-pathway-commons-data"></a>
</h3>
<p>Users can load alternative pathway data from the <a href="http://www.pathwaycommons.org/" class="external-link">Pathway Commons</a> repository
using the <strong>paxtoolsr</strong> package from <a href="https://bioconductor.org/packages/release/bioc/html/paxtoolsr.html" class="external-link">Bioconductor</a>.
This pathway data represents an update to the Pathway Commons data used
in the original 2010 NetBox publication. Below is an example that makes
use of data from the <a href="http://www.reactome.org/" class="external-link">Reactome pathway
database</a>.</p>
<p><strong>NOTE:</strong> Downloaded data is automatically cached to
avoid unnecessary downloads.</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BioPAX/paxtoolsr" class="external-link">paxtoolsr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">filename</span> <span class="op">&lt;-</span> <span class="st">"PathwayCommons.8.reactome.EXTENDED_BINARY_SIF.hgnc.txt.gz"</span></span>
<span><span class="va">sif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/paxtoolsr/man/downloadPc2.html" class="external-link">downloadPc2</a></span><span class="op">(</span><span class="va">filename</span>, version <span class="op">=</span> <span class="st">"8"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Filter interactions for specific types</span></span>
<span><span class="va">interactionTypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/paxtoolsr/man/getSifInteractionCategories.html" class="external-link">getSifInteractionCategories</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">filteredSif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/paxtoolsr/man/filterSif.html" class="external-link">filterSif</a></span><span class="op">(</span><span class="va">sif</span><span class="op">$</span><span class="va">edges</span>, interactionTypes <span class="op">=</span> <span class="va">interactionTypes</span><span class="op">[[</span><span class="st">"BetweenProteins"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">filteredSif</span> <span class="op">&lt;-</span> <span class="va">filteredSif</span><span class="op">[</span><span class="op">(</span><span class="va">filteredSif</span><span class="op">$</span><span class="va">INTERACTION_TYPE</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="st">"in-complex-with"</span><span class="op">)</span>,</span>
<span>    <span class="op">]</span></span>
<span></span>
<span><span class="co"># Re-run NetBox algorithm with new network</span></span>
<span><span class="va">graphReduced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/networkSimplify.html">networkSimplify</a></span><span class="op">(</span><span class="va">filteredSif</span>, directed <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">netbox2010</span><span class="op">$</span><span class="va">geneList</span><span class="op">)</span></span>
<span></span>
<span><span class="va">threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">pcResults</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/geneConnector.html">geneConnector</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">geneList</span>, networkGraph <span class="op">=</span> <span class="va">graphReduced</span>, directed <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    pValueAdj <span class="op">=</span> <span class="st">"BH"</span>, pValueCutoff <span class="op">=</span> <span class="va">threshold</span>, communityMethod <span class="op">=</span> <span class="st">"lec"</span>, keepIsolatedNodes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the p-value of the selected linker</span></span>
<span><span class="va">linkerDF</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">neighborData</span></span>
<span><span class="va">linkerDF</span><span class="op">[</span><span class="va">linkerDF</span><span class="op">$</span><span class="va">pValueFDR</span> <span class="op">&lt;</span> <span class="va">threshold</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># The geneConnector function returns a list of data frames.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot graph with the Fruchterman-Reingold layout algorithm</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">netboxCommunity</span>, <span class="va">results</span><span class="op">$</span><span class="va">netboxGraph</span>, layout <span class="op">=</span> <span class="va">layout_with_fr</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="selecting-input-gene-lists-for-use-with-netbox">Selecting Input Gene Lists for use with NetBox<a class="anchor" aria-label="anchor" href="#selecting-input-gene-lists-for-use-with-netbox"></a>
</h2>
<p>The main input for the NetBox algorithm is an input list of
“significantly” altered genes. Each project is different, unique
considerations for how significance should be considered may be
required. Researchers may seek stronger thresholds of significance for
particular questions and different profiling technologies may have their
own considerations. It is beyond the scope of this work to provide
guidance for all situations.</p>
<p>However, to help users better understand the process of generating an
input gene list we provide examples using best practices derived from
the The Cancer Genome Project using the cBioPortal (<a href="http://cbioportal.org/" class="external-link uri">http://cbioportal.org/</a>), a
platform that aggregates clinical genomics datasets into a standard
representation. As of August 2020, cBioPortal has approximately 290
studies. In cases where appropriate data is available a similar
procedure to the example can be used.</p>
<div class="section level3">
<h3 id="accesing-pre-computed-alteration-results-from-the-cbioportal-datahub">Accesing Pre-Computed Alteration Results from the cBioPortal
DataHub<a class="anchor" aria-label="anchor" href="#accesing-pre-computed-alteration-results-from-the-cbioportal-datahub"></a>
</h3>
<p>For TCGA studies on cBioPortal, users can access pre-processed
datasets from the <a href="https://github.com/cBioPortal/datahub/tree/master/public" class="external-link">cBioPortal
DataHub</a> that contain significantly altered genes by mutations and
copy number. Example study link: <a href="https://github.com/cBioPortal/datahub/tree/master/public/acc_tcga" class="external-link uri">https://github.com/cBioPortal/datahub/tree/master/public/acc_tcga</a></p>
<ul>
<li>Significantly altered genes by mutations (via MutSig algorithm) are
accessible within the ‘data_mutsig.txt’ file for a study; typically
mutations with a q-value &lt; 0.1 are selected as significantly
altered</li>
<li>Significantly altered genes by copy number (via GISTIC algorithm)
are accessible within the ‘data_gistic_genes_del.txt’ (deletions) file
and ‘data_gistic_genes_amp.txt’ (amplifications).</li>
</ul>
<p>Users are directed to the accompanying study publications; study
publication details are in the ‘meta_study.txt’ file for a study.</p>
</div>
<div class="section level3">
<h3 id="accessing-cancer-genomics-data-from-cbioportal">Accessing Cancer Genomics Data from cBioPortal<a class="anchor" aria-label="anchor" href="#accessing-cancer-genomics-data-from-cbioportal"></a>
</h3>
<p>Users can download cancer alteration data from <a href="https://www.cbioportal.org/" class="external-link">cBioPortal</a> using the
<strong>cgdsr</strong> package from <a href="https://cran.r-project.org/web/packages/cgdsr/index.html" class="external-link">CRAN</a>.
Here we show how a simple example for selecting genes for use with
netboxr for datasets provided by cBioPortal using a using a 10%
alteration frequency threshold to select genes; this general procedure
has previously been used as part of <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3910500/" class="external-link">TCGA
studies</a>. In the example, we consider:</p>
<ul>
<li>For mutations, mutations of any type contribute to the overall
alteration frequency of the gene</li>
<li>For copy number, discretized GISTIC-derived values for amplification
or deep deletions contribute to the overall alteration frequency</li>
</ul>
<p>The resulting gene list then becomes an input for netboxr. The
resulting gene list will select EGFR and TP53, which have high
alteration frequencies in glioblastoma (GBM) over the housekeeping genes
ACTB and GAPDH, which have very low alteration frequencies.</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cBioPortalData</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cbio</span> <span class="op">&lt;-</span> <span class="fu">cBioPortal</span><span class="op">(</span>hostname <span class="op">=</span> <span class="st">"www.cbioportal.org"</span>, protocol <span class="op">=</span> <span class="st">"https"</span>, api. <span class="op">=</span> <span class="st">"/api/api-docs"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find available studies, caselists, and geneticProfiles</span></span>
<span><span class="va">studies</span> <span class="op">&lt;-</span> <span class="fu">getStudies</span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></span>
<span><span class="va">samps</span> <span class="op">&lt;-</span> <span class="fu">sampleLists</span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"gbm_tcga_pub"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># find samples with both mutation and copy number data</span></span>
<span><span class="va">caseList</span> <span class="op">&lt;-</span> <span class="st">"gbm_tcga_pub_cnaseq"</span></span>
<span><span class="va">geneticProfileTables</span> <span class="op">&lt;-</span> <span class="fu">molecularProfiles</span><span class="op">(</span>api <span class="op">=</span> <span class="va">cbio</span>, studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"EGFR"</span>, <span class="st">"TP53"</span>, <span class="st">"RB1"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">genes</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"Work on %s gene"</span>, <span class="va">gene</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">cna</span> <span class="op">&lt;-</span> <span class="fu">getDataByGenes</span><span class="op">(</span><span class="va">cbio</span>, studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>, genes <span class="op">=</span> <span class="va">gene</span>, by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>        molecularProfileId <span class="op">=</span> <span class="st">"gbm_tcga_pub_cna_consensus"</span>, sampleListId <span class="op">=</span> <span class="va">caseList</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">mut</span> <span class="op">&lt;-</span> <span class="fu">getDataByGenes</span><span class="op">(</span><span class="va">cbio</span>, studyId <span class="op">=</span> <span class="st">"gbm_tcga_pub"</span>, genes <span class="op">=</span> <span class="va">gene</span>, by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>,</span>
<span>        molecularProfileId <span class="op">=</span> <span class="st">"gbm_tcga_pub_mutations"</span>, sampleListId <span class="op">=</span> <span class="va">caseList</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">cna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">cna</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span>, <span class="va">cna</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="va">mut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">mut</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>, <span class="va">mut</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">14</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">cna</span>, <span class="va">mut</span>, by <span class="op">=</span> <span class="st">"sampleId"</span>, all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">cna</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">$</span><span class="va">value</span></span>
<span></span>
<span>    <span class="va">mut</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">$</span><span class="va">proteinChange</span></span>
<span></span>
<span>    <span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>cna <span class="op">=</span> <span class="va">cna</span>, mut <span class="op">=</span> <span class="va">mut</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>    <span class="va">tmp</span><span class="op">$</span><span class="va">isAltered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">$</span><span class="va">cna</span><span class="op">)</span> <span class="op">==</span> <span class="fl">2</span> <span class="op">|</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">$</span><span class="va">mut</span><span class="op">)</span>  <span class="co"># Amplification or Deep Deletion or any mutation</span></span>
<span>    <span class="va">freq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">$</span><span class="va">isAltered</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span>
<span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">freq</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span>, USE.NAMES <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 10 percent alteration frequency cutoff</span></span>
<span><span class="va">geneList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span><span class="op">[</span><span class="va">results</span> <span class="op">&gt;</span> <span class="fl">0.1</span><span class="op">]</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Cerami E, Demir E, Schultz N, Taylor BS, Sander C (2010) Automated
Network Analysis Identifies Core Pathways in Glioblastoma. PLoS ONE
5(2): e8918. <a href="doi:10.1371/journal.pone.0008918" class="uri">doi:10.1371/journal.pone.0008918</a>
</li>
<li>Cerami EG, Gross BE, Demir E, Rodchenkov I, Babur O, Anwar N,
Schultz N, Bader GD, Sander C. Pathway Commons, a web resource for
biological pathway data. Nucleic Acids Res. 2011 Jan;39(Database
issue):D685-90. <a href="doi:10.1093/nar/gkq1039" class="uri">doi:10.1093/nar/gkq1039</a>. Epub 2010 Nov 10.</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] org.Hs.eg.db_3.17.0   AnnotationDbi_1.62.1  IRanges_2.34.0       </span></span>
<span><span class="co">##  [4] S4Vectors_0.38.1      Biobase_2.60.0        BiocGenerics_0.46.0  </span></span>
<span><span class="co">##  [7] clusterProfiler_4.8.1 RColorBrewer_1.1-3    netboxr_1.7.2        </span></span>
<span><span class="co">## [10] igraph_1.4.2          knitr_1.42            BiocStyle_2.28.0     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] jsonlite_1.8.4          magrittr_2.0.3          farver_2.1.1           </span></span>
<span><span class="co">##   [4] rmarkdown_2.21          fs_1.6.2                zlibbioc_1.46.0        </span></span>
<span><span class="co">##   [7] ragg_1.2.5              vctrs_0.6.2             memoise_2.0.1          </span></span>
<span><span class="co">##  [10] RCurl_1.98-1.12         ggtree_3.8.0            htmltools_0.5.5        </span></span>
<span><span class="co">##  [13] gridGraphics_0.5-1      sass_0.4.6              KernSmooth_2.23-20     </span></span>
<span><span class="co">##  [16] bslib_0.4.2             htmlwidgets_1.6.2       desc_1.4.2             </span></span>
<span><span class="co">##  [19] plyr_1.8.8              cachem_1.0.8            lifecycle_1.0.3        </span></span>
<span><span class="co">##  [22] pkgconfig_2.0.3         gson_0.1.0              Matrix_1.5-4           </span></span>
<span><span class="co">##  [25] R6_2.5.1                fastmap_1.1.1           GenomeInfoDbData_1.2.10</span></span>
<span><span class="co">##  [28] digest_0.6.31           aplot_0.1.10            enrichplot_1.20.0      </span></span>
<span><span class="co">##  [31] colorspace_2.1-0        patchwork_1.1.2         rprojroot_2.0.3        </span></span>
<span><span class="co">##  [34] crosstalk_1.2.0         textshaping_0.3.6       RSQLite_2.3.1          </span></span>
<span><span class="co">##  [37] labeling_0.4.2          fansi_1.0.4             httr_1.4.6             </span></span>
<span><span class="co">##  [40] polyclip_1.10-4         compiler_4.3.0          bit64_4.0.5            </span></span>
<span><span class="co">##  [43] withr_2.5.0             downloader_0.4          BiocParallel_1.34.1    </span></span>
<span><span class="co">##  [46] viridis_0.6.3           DBI_1.1.3               highr_0.10             </span></span>
<span><span class="co">##  [49] ggforce_0.4.1           gplots_3.1.3            MASS_7.3-58.4          </span></span>
<span><span class="co">##  [52] HDO.db_0.99.1           gtools_3.9.4            caTools_1.18.2         </span></span>
<span><span class="co">##  [55] tools_4.3.0             ape_5.7-1               scatterpie_0.1.9       </span></span>
<span><span class="co">##  [58] glue_1.6.2              nlme_3.1-162            GOSemSim_2.26.0        </span></span>
<span><span class="co">##  [61] shadowtext_0.1.2        grid_4.3.0              reshape2_1.4.4         </span></span>
<span><span class="co">##  [64] fgsea_1.26.0            generics_0.1.3          gtable_0.3.3           </span></span>
<span><span class="co">##  [67] tidyr_1.3.0             data.table_1.14.8       tidygraph_1.2.3        </span></span>
<span><span class="co">##  [70] utf8_1.2.3              XVector_0.40.0          ggrepel_0.9.3          </span></span>
<span><span class="co">##  [73] pillar_1.9.0            stringr_1.5.0           yulab.utils_0.0.6      </span></span>
<span><span class="co">##  [76] splines_4.3.0           dplyr_1.1.2             tweenr_2.0.2           </span></span>
<span><span class="co">##  [79] treeio_1.24.0           lattice_0.21-8          bit_4.0.5              </span></span>
<span><span class="co">##  [82] tidyselect_1.2.0        GO.db_3.17.0            Biostrings_2.68.1      </span></span>
<span><span class="co">##  [85] gridExtra_2.3           bookdown_0.34           xfun_0.39              </span></span>
<span><span class="co">##  [88] graphlayouts_1.0.0      DT_0.27                 stringi_1.7.12         </span></span>
<span><span class="co">##  [91] lazyeval_0.2.2          ggfun_0.0.9             yaml_2.3.7             </span></span>
<span><span class="co">##  [94] evaluate_0.21           codetools_0.2-19        ggraph_2.1.0           </span></span>
<span><span class="co">##  [97] tibble_3.2.1            qvalue_2.32.0           BiocManager_1.30.20    </span></span>
<span><span class="co">## [100] ggplotify_0.1.0         cli_3.6.1               systemfonts_1.0.4      </span></span>
<span><span class="co">## [103] munsell_0.5.0           jquerylib_0.1.4         Rcpp_1.0.10            </span></span>
<span><span class="co">## [106] GenomeInfoDb_1.36.0     png_0.1-8               ellipsis_0.3.2         </span></span>
<span><span class="co">## [109] pkgdown_2.0.7           ggplot2_3.4.2           blob_1.2.4             </span></span>
<span><span class="co">## [112] DOSE_3.26.1             bitops_1.0-7            viridisLite_0.4.2      </span></span>
<span><span class="co">## [115] tidytree_0.4.2          scales_1.2.1            purrr_1.0.1            </span></span>
<span><span class="co">## [118] crayon_1.5.2            rlang_1.1.1             cowplot_1.1.1          </span></span>
<span><span class="co">## [121] fastmatch_1.1-3         KEGGREST_1.40.0         formatR_1.14</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Eirc Minwei Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
